elastic/apm-server
# Remove source map upload endpoint, add support for fetching source maps from Kibana

## Motivation/summary

Remove the deprecated source map upload endpoint from APM Server. Introduce a new source map fetcher for standalone APM Server to fetch source maps directly from Kibana (requires `apm-server.kibana` config).


## Checklist

~- [ ] Documentation has been updated~ Opened https://github.com/elastic/apm-server/issues/6446

## How to test these changes

Test 1 (check source maps uploaded via Kibana):

1. Upload a source map through Kibana (https://www.elastic.co/guide/en/kibana/current/rum-sourcemap-api.html#rum-sourcemap-post)
2. Send some RUM spans or errors which should match the source map
3. Check that the indexed documents have source mapping applied

Test 2 (check source maps uploaded via [older] apm-server):

1. Attempt to upload source map through apm-server, check that the server returns a 404 (route does not exist any more)
2. Upload a source map through apm-server 7.16.0
3. Send some RUM spans or errors which should match the source map
4. Check that the indexed documents have source mapping applied

## Related issues

Closes #5993
